server/game/core/card/propertyMixins/UnitProperties.ts

```diff
@@ -52,6 +53,18 @@ export function WithUnitProperties<TBaseClass extends InPlayCardConstructor>(Bas
             return this._upgrades;
         }
 
+        public hasUpgradeWithName(name: string): boolean {
```

Comment:
Can we just remove this if it's not being used? It's kind of an anti-pattern we don't want to provide support for